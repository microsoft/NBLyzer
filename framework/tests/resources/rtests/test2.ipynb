{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","trusted":true},"outputs":[],"source":["# This Python 3 environment comes with many helpful analytics libraries installed\n","# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n","# For example, here's several helpful packages to load\n","\n","import numpy as np # linear algebra\n","import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n","\n","# Input data files are available in the read-only \"../input/\" directory\n","# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n","\n","import os\n","for dirname, _, filenames in os.walk('/kaggle/input'):\n","    for filename in filenames:\n","        print(os.path.join(dirname, filename))\n","\n","# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n","# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"79c7e3d0-c299-4dcb-8224-4455121ee9b0","_uuid":"d629ff2d2480ee46fbb7e2d37f6b5fab8052498a","trusted":true},"outputs":[],"source":["import numpy as np\n","import matplotlib.pyplot as plt\n","import pandas as pd\n","import seaborn as sns"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["df = pd.read_csv('/kaggle/input/heart-disease-uci/heart.csv')"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["df.isnull().sum()"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["df"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["sns.barplot(x=df['age'],y=df['target'])"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["plt.hist(x=df['age'],histtype='bar')"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["plt.hist(x=df['thalach'],histtype='bar',color='green')"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["plt.hist(x=df['chol'],histtype='bar',color='yellow')"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["from sklearn.preprocessing import StandardScaler , LabelEncoder\n","scaler = StandardScaler()"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["label = LabelEncoder()\n","train = df.iloc[:,:-1]\n","train"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["train['oldpeak'] = label.fit_transform(train['oldpeak'])"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["target = df['target']"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["X_scaled = scaler.fit_transform(train)"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["from sklearn.metrics import roc_curve , roc_auc_score , confusion_matrix , accuracy_score\n","from sklearn.model_selection import train_test_split\n","from sklearn.ensemble import GradientBoostingClassifier\n","grad = GradientBoostingClassifier()\n","X_train,X_test,y_train,y_test = train_test_split(X_scaled,target,test_size=0.4,random_state=120)\n","grad.fit(X_train,y_train)\n","y_pred=grad.predict(X_test)\n","accuracy_score(y_pred,y_test)"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["auc = roc_auc_score(y_test,y_pred)\n","confusion = confusion_matrix(y_test,y_pred)\n","tp = confusion[0][0]\n","fp = confusion[0][1]\n","fn = confusion[1][0]\n","tn = confusion[1][1]\n"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["from sklearn.metrics import plot_confusion_matrix\n","disp = plot_confusion_matrix(grad,X_test,y_test,cmap=plt.cm.Blues,normalize=None)\n","disp.confusion_matrix"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["\n","# finding accuracy \n","accuracy = (tp+tn)/(tp+tn+fp+fn)\n","accuracy"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["fpr , tpr , thresholds = roc_curve(y_test,y_pred)\n","plt.plot(fpr,tpr,color = 'darkblue',label = 'ROC')\n","plt.plot([0,1],[0,1],color='orange',linestyle='--',label=\"ROC Curve(area=%0.2f)\"%auc)\n","plt.xlabel('False + ve rate')\n","plt.ylabel('True +ve rate')\n","plt.legend()\n","plt.show()"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":[]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.6.4"}},"nbformat":4,"nbformat_minor":4}
