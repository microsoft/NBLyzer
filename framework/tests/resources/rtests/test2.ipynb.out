[{"cell_id":0,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Isolated cell."}],"path":[0]},{"cell_id":1,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Isolated cell."}],"path":[1]},{"cell_id":19,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Isolated cell."}],"path":[19]},{"cell_id":0,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[0]},{"cell_id":1,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[1]},{"cell_id":3,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[3]},{"cell_id":4,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[4]},{"cell_id":5,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[5]},{"cell_id":6,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[6]},{"cell_id":7,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[7]},{"cell_id":8,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[8]},{"cell_id":11,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[11]},{"cell_id":16,"errors":[{"line":2,"label":"disp", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[16]},{"cell_id":17,"errors":[{"line":1,"label":"accuracy", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[17]},{"cell_id":18,"errors":[{"line":1,"label":"fpr", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":1,"label":"tpr", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":1,"label":"thresholds", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[18]},{"cell_id":19,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[19]}]